src/assets/scripts/client/aircraft/AircraftModel.js

```diff
@@ -2753,4 +2753,30 @@ export default class AircraftModel {
             this.setIsFlightStripRemovable();
         }
     }
+
+    /**
+     * Ensure that the provided altitude is valid
+     *
+     * @for AircraftModel
+     * @method validateNextAltitude
+     * @param nextAltitude {number} altitude the aircraft should maintain
+     * @return {array|undefined}           [success of operation, readback]
+     */
+    validateNextAltitude(nextAltitude) {
+        if (nextAltitude === INVALID_NUMBER) {
+            return [false, 'unable, no altitude assigned'];
+        }
+
+        if (typeof nextAltitude !== 'number') {
+            return [false, `unable to maintain an altitude of ${nextAltitude}`];
+        }
+
+        if (!this.model.isAbleToMaintainAltitude(nextAltitude)) {
+            const readback = {};
+            readback.log = `unable to maintain ${nextAltitude} due to performance`;
+            readback.say = `unable to maintain ${radio_altitude(nextAltitude)} due to performance`;
+
+            return [false, readback];
+        }
```

Comment:
continuing off the last comment, this method should always return the same value type.  please add:
```js
return [];
```

Reply:
done